command/server.go

```diff
@@ -351,6 +353,13 @@ func (c *ServerCommand) Run(args []string) int {
 		}
 	}
 
+	// Make sure we close all listeners from this point on
+	defer func() {
```

Comment:
We have several paths by which the program can exit -- if it's in `verify-only` mode, normal termination, or some error condition leading to a `return`. So this way ensures that we close all of them to avoid leaking descriptors.